packages/components/src/autocomplete/index.js

```diff
@@ -268,11 +269,11 @@ function useAutocomplete( {
 		event.preventDefault();
 	}
 
-	let textContent;
-
-	if ( isCollapsed( record ) ) {
-		textContent = getTextContent( slice( record, 0 ) );
-	}
+	const textContent = useMemo( () => {
+		if ( isCollapsed( record ) ) {
+			return getTextContent( slice( record, 0 ) );
+		}
+	}, [ record ] );
```

Comment:
Sorry, I'm probably missing the context here, but I'm curious. What's the reason for using `memo` for the string value? The `getTextContent` doesn't look like an expensive method.

Reply:
To be honest, there's no particular reason for this one but my idea was that this component is kind of a bottleneck to the less we do on rendering, the better.

Reply:
That makes sense. Thank you, Riad 🙇

Reply:
Should we add some inline comment on these preemptive hot-path optimizations in case the opposite becomes true (and running useMemo becomes more expensive)?

Reply:
I can do that 👍